scripts/update_version.py

```
@@ -0,0 +1,623 @@
+
+
+class VersionUpdater:
+    def __init__(self, base_path: str):
+        self.base_path = base_path
+        
+    def get_current_versions(self) -> Dict[str, str]:
+        """Get current version numbers from the repository."""
+        versions = {}
+        
+        # Main API version
+        with open(os.path.join(self.base_path, "pyproject.toml"), "r") as f:
```

Ruff (0.11.9)

Unnecessary mode argument

Remove mode argument

(UP015)